src/main/java/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package strategy.impl;
+
+import strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int apply(int quantity) {
+        return quantity * (-1);
+    }
```

Comment:
The `PurchaseOperation` class does not follow the checklist requirement for handling the Purchase operation. It should check the result balance before saving it in the Storage and should throw a RuntimeException if the balance is negative. Currently, it just multiplies the quantity by -1, which does not ensure a positive balance in the storage.

Reply:
In this class we don't know with which product it makes purchase. I did this check in ShopServiceImpl class:
} else {
                **if (getBalance(transaction) + getHowManyHaveChanged(transaction) >= 0)** {
                    Storage.reports.put(transaction.getFruit(), getBalance(transaction)
                            + getHowManyHaveChanged(transaction));
                } else {
                    **throw new RuntimeException("The quantity at shop is not enough to purchase");**
                }